### PR TITLE
Rework candidate handling.

### LINKED ISSUE #64: Handle all ball candidates
Currently, only the top candidate gets added to the ball candidates message, but the message is designed to contain all candidates.

### LINKED ISSUE #131: Check for detected goal, if on green field
Currently, there is no check to filter for FalsePositive detected goals.
Check for each detected goal, if on green field.

### PR DESCRIPTION
**Pull request by Flova**
This pull request reworks the candidate handling. This makes the `BallDetector` obsolete. It is now also more flexible and allows us to sort out goal posts that are too far away from the field boundary.

## Proposed changes
- Remove `BallDetector`
- Rename `DummyBallDetector` as `DummyCandidateFinder`
- Move `DummyCandidateFinder` to `candidate.py` file 
- Change sorting and field boundary filtering
- Filter goal posts

## Related issues
Closes #131 
Closes #64 

## Necessary checks
- [x] Update package version
- [x] Run linters
- [x] Run `catkin build`
- [x] Write documentation
- [x] Test on your machine
- [ ] ~~Test on the robot~~

### CODE REVIEW

File: bitbots_vision/src/bitbots_vision/vision_modules/ros_utils.py [hunk 1]
```diff
@@ -286,8 +286,8 @@ def build_goal_msg(goal_parts_msg):
     Builds a goal message with a right and left post. If there is only one post in the image, the right and left post are the same.
     This should be reworked! The vision should only publish posts and e.g. the worldmodel builds a goal out of this context.
 
-    :param top_ball_candidate: best rated ball candidate
-    :return: ball msg
+    :param goal_parts_msg: goal parts as ros message
+    :return: goal message
```
Comment (JanGut) [inline bitbots_vision/src/bitbots_vision/vision_modules/ros_utils.py]: Please add type descriptions

**Comment by JanGut** [comment]
Please test this on a mashine.

File: bitbots_vision/src/bitbots_vision/vision.py [hunk 2]
```diff
@@ -151,13 +151,18 @@ def _configure_vision(self, config, level):
         """
         self._register_or_update_all_publishers(config)
 
+        # Set max number of balls
+        self._max_balls = 1
```
[the remark this answers is not in the document]
Comment (Flova) [reply bitbots_vision/src/bitbots_vision/vision.py]: Done

File: bitbots_vision/src/bitbots_vision/vision_modules/ros_utils.py (hunk 1 above)
Comment (Flova) [reply bitbots_vision/src/bitbots_vision/vision_modules/ros_utils.py]: I think this is a general problem and does not belong to this pr. Maybe we should open an issue for this topic.

**Comment by timonegk** [comment]
Please update the package version.

File: bitbots_vision/src/bitbots_vision/vision_modules/yolo_handler.py [hunk 3]
```diff
@@ -7,7 +7,7 @@
 except ImportError:
     rospy.logerr("Not able to run Darknet YOLO! Its only executable under python3 with yolo34py or yolo34py-gpu installed.", logger_name="vision_yolo")
 import numpy as np
-from .candidate import CandidateFinder, BallDetector, Candidate
+from .candidate import CandidateFinder, CandidateFinder, Candidate
```
Comment (JanGut) [inline bitbots_vision/src/bitbots_vision/vision_modules/yolo_handler.py]:
```suggestion
from .candidate import CandidateFinder, Candidate
```
As suggested by code factor, please resolve this.

**Pull request merged**